pkg/credentialprovider/gcp/metadata.go

```diff
@@ -121,7 +121,8 @@ func onGCEVM() bool {
 		glog.V(2).Infof("Error while reading product_name: %v", err)
 		return false
 	}
-	return strings.Contains(string(data), googleProductName)
+	name := strings.TrimSpace(string(data))
+	return name == "Google" || name == "Google Compute Engine"
```

Comment:
I wonder if "Google" is used anywhere as I've only seen "Google Compute Engine".

Reply:
@yujuhong see the picture of the device and details - https://github.com/kubernetes/kubernetes/issues/57760#issuecomment-356466614

Reply:
I saw the picture. The question is whether there exists a file where the full product name == "Google" :-)

Reply:
Aha. :)